#! /bin/sh -e
## 313_pam_access_with_preauth.dpatch.dpatch by Bjorn Torkelsson <Bjorn.Torkelsson@hpc2n.umu.se>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Allow PAM account authorization when preauthenticated
## DP: Debian bug #193869

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

@DPATCH@

--- shadow-4.0.3/src/login.c.orig	2004-08-04 11:03:14.967848591 +0200
+++ shadow-4.0.3/src/login.c	2004-08-04 11:04:43.950147350 +0200
@@ -895,6 +895,11 @@
 		setpwent ();
 		pwd = getpwnam (pam_user);
 
+		if (fflg) {
+			retcode = pam_acct_mgmt(pamh, 0);
+			PAM_FAIL_CHECK;
+		}
+
 		if (!pwd || setup_groups (pwd))
 			exit (1);
 		else
